### PR TITLE
Update isolated analysis in the init method

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/40736

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by chiranSachintha** [comment]
Consider the following example. Is it correct to access `arr` without a lock inside the lambda function? The following program compile without any errors.
```ballerina
isolated class IsolatedClass {
    private int[][] arr;

    function init(int[] node) {
        self.arr = [];
        function v = function () {
            self.arr.push(node);
        };
    }
}
```

**Comment by MaryamZi** [comment]
> Consider the following example. Is it correct to access `arr` without a lock inside the lambda function? The following program compile without any errors.
> 
> ```ballerina
> isolated class IsolatedClass {
>     private int[][] arr;
> 
>     function init(int[] node) {
>         self.arr = [];
>         function v = function () {
>             self.arr.push(node);
>         };
>     }
> }
> ```

Seems like there is an issue here, but is also the case in non-`init` methods. Can you please create an issue for this?

**Review by chiranSachintha — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java [hunk 1]
```diff
@@ -2612,6 +2615,38 @@ private boolean isInvalidIsolatedObjectFieldOrMethodAccessViaSelfIfOutsideLock(
...
+            parent = parent.parent;
+        }
+
+        if (parent == null) {
+            return false;
+        }
+
+        BLangAssignment assignment = (BLangAssignment) parent;
+
+        BLangExpression lhsExpr = assignment.varRef;
```
Comment (dulajdilshan) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java:2635]: We can remove the newline here. (Ignorable)

**Review by dulajdilshan — APPROVED** [review]
LGTM!. Only minor ignorable comment.

**Pull request merged**